File: ports/csharp/src/core/graphics/BcsGraphicsTests.cs
using System;
using System.Diagnostics;
using Bcs.Core.Kernel;

namespace Bcs.Core.Graphics.Tests
{
    public class BcsGraphicsTests
    {
        public static void TestGraphicsLifecycle()
        {
            var parent = new BcsObject();
            var canvas = new BcsCanvas(parent, 800, 600);
            var painter = new BcsPainter(parent);

            Debug.Assert(canvas.Parent() == parent);
            var (w, h) = canvas.Size();
            Debug.Assert(w == 800 && h == 600);

            Debug.Assert(painter.Begin(canvas));
            Debug.Assert(!painter.Begin(canvas));

            painter.End();
        }
    }
}

File: ports/java/src/core/graphics/BcsGraphicsTests.java
package core.graphics;

import core.kernel.BcsObject;

public class BcsGraphicsTests {
    public static void testGraphicsLifecycle() {
        BcsObject parent = new BcsObject(null);
        BcsCanvas canvas = new BcsCanvas(parent, 800, 600);
        BcsPainter painter = new BcsPainter(parent);

        if (canvas.getParent() != parent) {
            throw new AssertionError("Parent mismatch");
        }

        int[] size = canvas.getSize();
        if (size[0] != 800 || size[1] != 600) {
            throw new AssertionError("Size mismatch");
        }

        if (!painter.begin(canvas)) {
            throw new AssertionError("Begin failed");
        }

        if (painter.begin(canvas)) {
            throw new AssertionError("Double begin should fail");
        }

        painter.end();
    }
}
